### PR TITLE
FIX: Don't error when rotating identities of protocol-v0

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Users who are currently stuck with v0 identities are unable to rotate their identities into v1 because of a bug where the old identity is assumed to be v1 and has signing keys, but v0 identities don't have signing keys. When rotating an identity, the signing keys of the old identity are copied to the new identity to keep old signatures working. However, when the old identity is v0 (which has no signing keys), we attempt to copy the old signature keys (`undefined`) to the new v1 identity, but it ends up with no signing keys and fails to be saved because v1 identities are expected to have signing keys.

This PR fixes the issue by checking for the existence of old signing keys before copying them to the new identity. If there are no signing keys in the old identity, we let the new identity save with the signing keys that are generated with it.

### CODE REVIEW

File: assets/javascripts/lib/protocol.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ export const ENCRYPT_PROTOCOL_VERSION = 1;
  * @return {Promise}
  */
 export function generateIdentity(version) {
-  version = version || ENCRYPT_PROTOCOL_VERSION;
+  if (version === undefined) {
+    version = ENCRYPT_PROTOCOL_VERSION;
+  }
```
Comment (jjaffeux) [inline assets/javascripts/lib/protocol.js:48]: Why this change?

Could also probably do: 

```
version ??= ENCRYPT_PROTOCOL_VERSION;
```

**Review by jjaffeux — APPROVED** [review]

File: assets/javascripts/lib/protocol.js (hunk 1 above)
Comment (OsamaSayegh) [reply assets/javascripts/lib/protocol.js]: Because it wasn't possible to generate a version `0` identity since the `||` treats `0` as a falsey value and it gets changed to 1. TIL about `??=` which is more suitable than `||` in this case, thanks!
